File: extract_bodyweight_from_emt.py
import os

def extract_bodyweight_from_emt(file_path):
    """
    Extract bodyweight from an EMT file in the specified directory
    
    Parameters:
        file_path (str): The path to the .emt file
    
    Returns:
        float: The extracted body weight in kilograms
    """
    # Open the file
    file_1 = next((f for f in os.listdir(file_path) if f.endswith('.emt')), None)
    if not file_1:
        raise FileNotFoundError("No .emt file found in the specified directory.")

    file = os.path.join(file_path, file_1)
    
    body_weight = None

    # Read the file line by line
    with open(file, 'r') as file_id:
        for line in file_id:
            if 'mTB' in line:
                body_weight = float(next(file_id).strip())
                break

    # Check if the body weight was found and handle the case where it was not
    if body_weight is None:
        raise ValueError("Body weight not found in the .emt file.")

    return body_weight

File: COMAK/matlab_scripts/extract_bodyweight_from_emt.m
function bodyWeight = extract_bodyweight_from_emt(filePath)
    % EXTRACT_BODYWEIGHT_FROM_EMT Extracts body weight from an EMT file.
    %
    % This function reads an EMT file in the specified directory and extracts the 
    % body weight value associated with the "mTB" line.
    %
    % Parameters:
    %   filePath (string): The path to the directory containing the .emt file.
    %
    % Returns:
    %   bodyWeight (double): The extracted body weight in kilograms.
    %
    % Author: Aurel Berger
    % Date: July 2024

    % Open the file
    file_1 = dir(fullfile(filePath, '*.emt'));
    file = [file_1.folder '/' file_1.name];

    fileID = fopen(file, 'r');
    if fileID == -1
        error('Cannot open the file: %s', file_1.name);
    end

    % Initialize body weight variable
    bodyWeight = [];

    % Read the file line by line
    line = fgetl(fileID);
    while ischar(line)
        % Check for the "mTB" line and extract the body weight
        if contains(line, 'mTB')
            line = fgetl(fileID);  % Read the next line containing the weight
            bodyWeight = str2double(strtrim(line));
            break;
        end
        line = fgetl(fileID);
    end

    % Close the file
    fclose(fileID);

    % Check if the body weight was found and handle the case where it was not
    if isempty(bodyWeight)
        error('Body weight not found in the .emt file.');
    end
end
